core/domain/activity_services_test.py

```diff
@@ -0,0 +1,201 @@
+        rights_manager.publish_collection(self.owner_id, self.COL_ID_2)
+
+        self._compare_lists(
+            activity_services.get_featured_activity_references(), [])
+
+        activity_services.update_featured_activity_references([
+            self._get_exploration_reference(self.EXP_ID_0),
+            self._get_collection_reference(self.COL_ID_2)])
+        self._compare_lists(
+            activity_services.get_featured_activity_references(), [
+                self._get_exploration_reference(self.EXP_ID_0),
+                self._get_collection_reference(self.COL_ID_2)])
+
+        activity_services.update_featured_activity_references([])
+        self._compare_lists(
+            activity_services.get_featured_activity_references(), [])
+
+    def test_error_handling(self):
+        rights_manager.publish_exploration(self.owner_id, self.EXP_ID_0)
+        rights_manager.publish_collection(self.owner_id, self.COL_ID_2)
+        self._compare_lists(
+            activity_services.get_featured_activity_references(), [])
+
+        with self.assertRaisesRegexp(Exception, 'should not have duplicates'):
+            activity_services.update_featured_activity_references([
+                self._get_exploration_reference(self.EXP_ID_0),
+                self._get_exploration_reference(self.EXP_ID_0)])
+
+    def test_deleted_activity_is_removed_from_featured_list(self):
+        rights_manager.publish_exploration(self.owner_id, self.EXP_ID_0)
+        rights_manager.publish_exploration(self.owner_id, self.EXP_ID_1)
+        rights_manager.publish_collection(self.owner_id, self.COL_ID_2)
+        activity_services.update_featured_activity_references([
+            self._get_exploration_reference(self.EXP_ID_0),
+            self._get_collection_reference(self.COL_ID_2)])
+
+        self._compare_lists(
+            activity_services.get_featured_activity_references(), [
+                self._get_exploration_reference(self.EXP_ID_0),
+                self._get_collection_reference(self.COL_ID_2)])
+
+        # Deleting an unfeatured activity does not affect the featured list.
+        exp_services.delete_exploration(self.owner_id, self.EXP_ID_1)
+        self._compare_lists(
+            activity_services.get_featured_activity_references(), [
+                self._get_exploration_reference(self.EXP_ID_0),
+                self._get_collection_reference(self.COL_ID_2)])
+
+        # Deleting a featured activity removes it from the featured list.
+        collection_services.delete_collection(self.owner_id, self.COL_ID_2)
+        self._compare_lists(
+            activity_services.get_featured_activity_references(), [
+                self._get_exploration_reference(self.EXP_ID_0)])
+        exp_services.delete_exploration(self.owner_id, self.EXP_ID_0)
+        self._compare_lists(
+            activity_services.get_featured_activity_references(), [])
+
+    def test_unpublished_activity_is_removed_from_featured_list(self):
```

Comment:
See `summary_services_test.ActivityReferenceValidationTests` -- I think that's what you mean?

Note that activity_services is meant to be 'lower' in the stack than summary_services, exp_services, etc. btw, so we can't do that check here, otherwise we'll introduce a circular dependency.